prompts/precision_v1.txt

```diff
@@ -0,0 +1,69 @@
+- `label`: A descriptive name for the term
+- `description`: A longer, possibly truncated description of what the term is; may be NA (i.e. absent)
+
+Here is a list of such records (in YAML format) where the `precision` label is already assigned for 3 examples at each level of precision:
+
+--- BEGIN EXAMPLES ---
+- id: EFO:1000639
+  label: acquired metabolic disease
+  definition: A disease of metabolism that has _material_basis_in enzyme deficiency or accumulation of enzymes or toxins which interfere with normal function due to an endocrine organ disease, organ malfunction, inadequate intake, dietary deficiency, or ...
+  precision: low
+- id: Orphanet:68336
+  label: Rare genetic tumor
+  definition: NA
+  precision: low
+- id: EFO:0005548
+  label: developmental disorder of mental health
+  definition: A disease of mental health that occur during a child’s developmental period between birth and age 18 resulting in retarding of the child’s
+  precision: low
+- id: EFO:0005548
+  label: inflammatory bowel disease
+  definition: A spectrum of small and large bowel inflammatory diseases of unknown etiology. It includes Crohn's disease, ulcerative colitis, and colitis of indeterminate type.
+  precision: medium
+- id: EFO:0000384
+  label: Crohn's disease
+  definition: A gastrointestinal disorder characterized by chronic inflammation involving all layers of the intestinal wall, noncaseating granulomas affecting the intestinal wall and regional lymph nodes, and transmural fibrosis. Crohn disease most ...
+  precision: medium
+- id: MONDO:0045020
+  label: glycine metabolism disease
+  definition: A disease that has its basis in the disruption of glycine metabolic process.
+  precision: medium
+- id: EFO:1000277
+  label: Gastric Small Cell Neuroendocrine Carcinoma
+  definition: An aggressive, high-grade and poorly differentiated carcinoma with neuroendocrine differentiation that arises from the stomach. It is characterized by the presence of malignant small cells.
+  precision: high
+- id: MONDO:0015634
+  label: isolated osteopoikilosis
+  definition: A osteopoikilosis (disease) that is not part of a larger syndrome.
+  precision: high
+- id: Orphanet:98755
+  label: Spinocerebellar ataxia type 1
+  definition: Spinocerebellar ataxia type 1 (SCA1) is a subtype of type I autosomal dominant cerebellar ataxia (ADCA type I; see this term) characterized by dysarthria, writing difficulties, limb ataxia, and commonly nystagmus and saccadic abnormalities.
+  precision: high
+--- END EXAMPLES ---
+
+Here are the records for which this `precision` label is not yet known:
+
+--- BEGIN RECORDS ---
+{records}
+--- END RECORDS ---
+
+Requirements:
+
+- Assign a `precision` label for ALL records
+- Respond in CSV format using a pipe (i.e. "|") delimiter with the headers `id`, `precision` where `id` is the `id` associated with each record
+- Include the headers in the result
+- Respond with ONLY the CSV content, do not include explanation of any kind
```

Comment:
Makes a lot of sense! As you know, it would make the parsing of the output a little bit more challenging but feasible.

One question for you:
I can see these chain-of-thought prompts having 2 main advantages:
1. Helps with troubleshooting, developing the prompt and introspecting performance
2. Helps give the model more tokens to spread it's reasoning 

Do you use chain-of-thought prompts most for 1. or also 2.?

Reply:
In my experience, I would use it mostly for 2 i.e. improve quality of responses by increasing token/reason. Clearly a nice side effect of 2 is 1. Some strategies we have explored wrt to parsing the results of longer/complicated completions:
 * be very explicit about the structure of the "final conclusions", e.g.: request conclusions in json. Copy-pasting from Eric:
 
 > FWIW, I suspect (especially after [this](https://github.com/eric-czech/ngly1-gpt/issues/1#issuecomment-1631809885)) that GPT4 understands jsonschema very well and you could probably avoid some post-hoc validation with something like:
>
```
The very last line of your response MUST be a single json object on a single line matching this schema:
"""
{
  "title": "Label",
  "type": "object",
  "properties": {
    "label": {
      "type": "string",
      "enum": ["low", "medium", "high"]
      "description": "..."
    }
  }
}
"""
```
 * alternatively you could use follow up request to extract specific data point from a previous completion, clearly this increases the cost/time, but you can use GPT-4 for the high quality reasoning, and GPT-3.5 for dummy extraction

Reply:
Makes a lot of sense! Let's keep this ^ for future work :)